lang/test/org/partiql/lang/eval/EvaluatingCompilerTests.kt

```diff
@@ -1475,4 +1535,68 @@ class EvaluatingCompilerTests : EvaluatorTestBase() {
                      listOf("a", "b"),
                      result.first().orderedNames)
     }
+
+    @Test
+    fun selectDistinct() = assertEval(
+        """SELECT DISTINCT t.a FROM `[{a: 1}, {a: 2}, {a: 1}]` t""",
+        """
+          [{a: 1}, {a: 2}]
+        """)
+
+    @Test
+    fun selectDistinctWithAggregate() = assertEval(
+        """SELECT SUM(DISTINCT t.a) AS a FROM `[{a:10}, {a:1}, {a:10}, {a:3}]` t""",
+        "[{a:14}]")
+
+    @Test
+    fun selectDistinctSubQuery() = assertEval(
+        """SELECT * FROM (SELECT DISTINCT t.a FROM `[{a: 1}, {a: 2}, {a: 1}]` t)""", 
+        """[{a:1},{a:2}]""")
+    
+    @Test
+    fun selectDistinctWithGroupBy() = assertEval(
+        """
+            SELECT t.a, COUNT(DISTINCT t.b) AS c 
+            FROM `[{a:1, b:10}, {a:1, b:10}, {a:1, b:20}, {a:2, b:10}, {a:2, b:10}]` t
+            GROUP by t.a
+        """,
+        """[{a:1, c:2}, {a:2, c:1}]""")
+
+    @Test
+    fun selectDistinctWithJoin() = assertEval(
```

Comment:
I believe if I add the unique values in `t1` and `t2` we cover this right? No need to add another test

Reply:
actually since distinct acts on the result of the cross join between `t1` and `t2` the current test already shows that we are picking values from both, `_1` is always from `t1` and `_2` is always from `t2`. 

If there is a mixup there it's bug on the cross join. Don't think we need to change this as it'd make the expected result much longer (from 4 to 9 elements) and harder to identify what went wrong if there is a regression